sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -6,28 +6,39 @@
 from pyro.infer.mcmc.api import MCMC
 from sbi.mcmc import Slice
 from torch import multiprocessing as mp
-
-from pyknos.nflows import flows
 from sbi.mcmc import SliceSampler
+import sbi.inference
 
 
-class FlowPosterior(flows.Flow):
+class Posterior:
     def __init__(
         self,
-        transform,
-        distribution,
+        algorithm,
+        neural_net,
         prior,
         context,
-        embedding=None,
-        train_with_mcmc=False,
+        train_with_mcmc=True,
         mcmc_method="slice-np",
     ):
+        """
+        Args:
+            algorithm: string, 'snpe', 'snl', 'sre'
```

Comment:
I'd make sure this is consistent with the class names, i.e., include "SnpeA", "SnpeB", "APT", ..